bgpd/bgp_rpki.c

```diff
@@ -131,12 +131,14 @@ static route_map_result_t route_match(void *rule, struct prefix *prefix,
 static void *route_match_compile(const char *arg);
 static void revalidate_bgp_node(struct bgp_node *bgp_node, afi_t afi,
 				safi_t safi);
+static void revalidate_all_routes(void);
 
 static struct rtr_mgr_config *rtr_config;
 static struct list *cache_list;
 static int rtr_is_running;
 static int rtr_is_stopping;
 static int rtr_is_starting;
+static _Atomic int rtr_update_overflow;
```

Comment:
As far as I can tell there's no pthreads being used here. This doesn't need to and should not be atomic, primarily because it's confusing and also because the cost of an atomic load / store is much higher for the memory orderings you chose. And I think if there were in fact two pthreads running the code that loads and the code that stores at the same time, you wouldn't get the desired behavior.

Reply:
Yes, rtrlib uses pthreads internally, and all callbacks it does are called from rtrlib's thread.